api/client/src/main/java/org/projectnessie/client/http/HttpClientBuilderImpl.java

```diff
@@ -77,13 +77,13 @@ final class HttpClientBuilderImpl implements HttpClient.Builder {
 
   HttpClientBuilderImpl(HttpClientBuilderImpl other) {
     this.baseUri = other.baseUri;
-    this.mapper = other.mapper;
+    this.mapper = other.mapper == null ? null : other.mapper.copy();
```

Comment:
Not related to the bug with the OAuth2 client, but I think `ObjectMapper` instances, while thread-safe, are not immutable. So it's probably safer to copy them as well.